spec/civiltime.html

```diff
@@ -1,230 +1,558 @@
+          1. Throw a *TypeError* exception.
+        1. Let _duration_ be ? ToTimeDurationLike(_duration_).
+        1. Let _hour_ be _time_.[[Hour]] - _duration_.[[Hour]].
+        1. Let _minute_ be _time_.[[Minute]] - _duration_.[[Minute]].
+        1. Let _second_ be _time_.[[Second]] - _duration_.[[Second]].
+        1. Let _millisecond_ be _time_.[[Millisecond]] - _duration_.[[Millisecond]].
+        1. Let _microsecond_ be _time_.[[Microsecond]] - _duration_.[[Microsecond]].
+        1. Let _nanosecond_ be _time_.[[Nanosecond]] - _duration_.[[Nanosecond]].
+        1. <mark>TODO: overflow</mark>
+        1. Return ? CreateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Time.prototype.with ( _timelike_ [ , _disambiguation_ ] )</h1>
+      <p>
+        The `with` method takes two arguments, _timelike_ and _disambiguation_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _time_ be the *this* value.
+        1. If Type(_time_) is not Object or _time_ does not have an [[InitializedTemporalTime]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _timelike_ be ? ToPartialTime(_timelike_).
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If _timelike_.[[Hour]] is not *undefined*, then
+          1. Let _hour_ be _timelike_.[[Hour]].
+        1. Else
+          1. Let _hour_ be _time_.[[Hour]].
+        1. If _timelike_.[[Minute]] is not *undefined*, then
+          1. Let _minute_ be _timelike_.[[Minute]].
+        1. Else
+          1. Let _minute_ be _time_.[[Minute]].
+        1. If _timelike_.[[Second]] is not *undefined*, then
+          1. Let _second_ be _timelike_.[[Second]].
+        1. Else
+          1. Let _second_ be _time_.[[Second]].
+        1. If _timelike_.[[Millisecond]] is not *undefined*, then
+          1. Let _millisecond_ be _timelike_.[[Millisecond]].
+        1. Else
+          1. Let _millisecond_ be _time_.[[Millisecond]].
+        1. If _timelike_.[[Microsecond]] is not *undefined*, then
+          1. Let _microsecond_ be _timelike_.[[Microsecond]].
+        1. Else
+          1. Let _microsecond_ be _time_.[[Microsecond]].
+        1. If _timelike_.[[Nanosecond]] is not *undefined*, then
+          1. Let _nanosecond_ be _timelike_.[[Nanosecond]].
+        1. Else
+          1. Let _nanosecond_ be _time_.[[Nanosecond]].
+        1. If ! ValidateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_) is *false*, then
+          1. Throw a *TypeError* exception.
+        1. If _disambiguation_ is not *undefined*, then
+          1. Set _disambiguation_ to ? ToString(_disambiguation_).
+        1. Else
+          1. Set _disambiguation_ to `"constrain"`.
+        1. Let _result_ be ? RegulateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_, _disambiguation_).
```

Comment:
this seems to be repeating a lot of the steps from the constructor. Can they be consolidated?

Reply:
Not all that much, actually. I think in this case it's better to leave the argument coercion to the method itself rather than trying to extract something here.